services/gitclub/app/routes/issues.py

```diff
@@ -1,64 +1,81 @@
 from flask import Blueprint, g, request, jsonify
 from werkzeug.exceptions import Forbidden, NotFound
 
-from ..models import Repo, Issue
-from .helpers import authorize, authorized_resources, oso
+from ..models import Repository, Issue, User
+from .helpers import actions, authorize, list_resources, object_to_typed_id, oso
 
 bp = Blueprint(
-    "routes.issues",
+    "issues",
     __name__,
     url_prefix="/orgs/<int:org_id>/repos/<int:repo_id>/issues",
 )
 
 
 @bp.route("", methods=["GET"])
 def index(org_id, repo_id):
-    repo = g.session.get_or_404(Repo, id=repo_id)
-    if not authorize("list_issues", repo):
+    repo = {"type": "Repository", "id": repo_id}
+    if not authorize("read", repo):
         raise NotFound
-    authorized_ids = authorized_resources("read", "Issue")
-    if authorized_ids and authorized_ids[0] == "*":
-        issues = g.session.query(Issue)
-        return jsonify([issue.repr() for issue in issues])
-    else:
-        issues = g.session.query(Issue).filter(Issue.id.in_(authorized_ids))
-        return jsonify([issue.repr() for issue in issues])
+    args = request.args
+    filters = []
+    # TODO: these aren't actually possible to set in
+    # the UI right now
+    if "is:open" in args:
+        filters.append(Issue.closed == False)
+    if "is:closed" in args:
+        filters.append(Issue.closed == True)
+    
+    issue_ids = list_resources("read", "Issue", repo_id)
+    issues = g.session.query(Issue).filter(Issue.repo_id == repo_id, *filters, Issue.id.in_(issue_ids)).order_by(Issue.issue_number)
+    return jsonify([issue.as_json() for issue in issues])
 
 
 @bp.route("", methods=["POST"])
 def create(org_id, repo_id):
+    if not authorize("read", {"type": "Repository", "id": repo_id}):
+        raise NotFound
     payload = request.get_json(force=True)
-    repo = g.session.get_or_404(Repo, id=repo_id)
+    repo = g.session.get_or_404(Repository, id=repo_id)
     if not authorize("create_issues", repo):
         raise NotFound
-    issue = Issue(title=payload["title"], repo=repo, creator_id=g.current_user.id)
+    issue = Issue(title=payload["title"], repo=repo, creator_id=g.current_user.username)
     g.session.add(issue)
     g.session.commit()
     oso.bulk_tell(
```

Comment:
I'm probably going to add a wrapper around this function, so that everything goes through `helpers` (soon to be called `authorization`).